kpet/data.py

```diff
@@ -296,6 +299,10 @@ def __init__(self, data):
             ),
             data
         )
+        if self.sets is None:
+            self.sets = set()
+        elif isinstance(self.sets, list):
+            self.sets = set(self.sets)
```

Comment:
I don't think we need to convert the list of regexes to a set at this time. Just keep it as a list until resolving to set name set. The code will be shorter, and we won't save much time by removing the very rare exact regex duplicates anyway.

Reply:
So. basically, we can just remove these four lines, I think :)

Reply:
You're right, this is a leftover from when I was converting the sets differently and I used some sets only operation on this attribute